peps/pep-0661.rst

```diff
@@ -197,6 +191,50 @@ add minimal complexity.
 
 Ordering comparisons are undefined for sentinel objects.
 
+Typing
+------
+
+To make usage of sentinels convenient in typed Python code, we propose to add
+amend the type system with a special case for sentinel objects.
+
+Sentinel objects may be used in :term:`type expressions <typing:type expression>`,
+representing themselves.
+This is similar to how ``None`` is handled in the existing type system.
+For example::
+
+    from sentinels import Sentinel
+
+    MISSING = Sentinel('MISSING')
+
+    def foo(value: int | MISSING = MISSING) -> int:
+        ...
+
+More formally, type checkers should recognize sentinel creations of the form
+``NAME = Sentinel('NAME')`` as creating a new sentinel objects. If the name
```

Comment:
Typo: "a new sentinel objects" should be "a new sentinel object" (singular).

Reply:
Thanks, fixed both